ui/admin/app/components/knowledge/AgentKnowledgePanel.tsx

```diff
@@ -107,7 +107,10 @@ export default function AgentKnowledgePanel({
 
     const getKnowledgeSources = useSWR(
         KnowledgeService.getKnowledgeSourcesForAgent.key(agentId),
-        ({ agentId }) => KnowledgeService.getKnowledgeSourcesForAgent(agentId),
+        ({ agentId }) =>
```

Comment:
when performing mutations on swr fetchers, please pull out the mutation into it's own useMemo.

If I were to try to use this key again with the generic fetcher method, my code would break because the data from the response would not align with the data in the cache

Reply:
example

```ts
// wrong

const { data }  = useSWR(
    Service.method.key(),
    () => Service.method().then(() => someMutation()) // cache has mutated data
)

// in another file

const { data }  = useSWR(
    Service.method.key(),
    () => Service.method() // cache has data directly from response
)

// an error may be thrown because the data in the cache does not match the structure of the data from the response
```

instead do this:

```ts
const { data }  = useSWR(
    Service.method.key(),
    () => Service.method() // use data directly from response
)

const mutatedData = useMemo(() => someMutation(data), [data])

// in another file

const { data }  = useSWR(
    Service.method.key(),
    () => Service.method() // also use data directly from response
)


No error now, because the data in the cache will ALWAYS have the same structure
```

Reply:
Good to know! changed 👍

Reply:
It's ugly, but it's important... If I ever get the time to replace SWR with react-query, this will be super trivial